tests: Fix alignment

eobalski opened the pull request:
@LixinGuoX could you verify if this fixes #44259?

Commit:
tests: Align buffer in icmsg_buf tests.

This change adds alignment to buffer on which icmsg_buf
is created. Some platforms run into fatal exception
when accessing unaligned addresses. Fix that by ensure
4 byte alignment.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

Commit:
tests: Move icmsg_buf tests to subsys/ipc

Move ipc service backend unit test from tests/lib to
tests/subsys directory.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

emob-nordic commented:
@aspeeddylan I saw you added
[default CACHE_TYPE as HAS_EXTERNAL_CACHE](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/aspeed/ast10x0/Kconfig.defconfig.series#L18:L20). In this PR the test that uses cache management is run and it fails due to undefined reference to `cache_data_range`.
```
[...]/ast1030_evb/tests/subsys/ipc/icmsg_buf/subsys.ipc.icmsg_buf/../../../../../../../include/cache.h:91: undefined reference to `cache_data_range'
```
Could you please verify if `HAS_EXTERNAL_CACHE` should be selected for this particular soc?
It doesn't work with `HAS_ARCH_CACHE` either.
Could you please have a look? Should i exclude this platform from this test?

nashif commented:
@emob-nordic this fixes the issue, but please fix the commit message: "s/a" is as bad as not having anything in the commit body.

aspeeddylan commented:
Hi Eobalski,

This SOC has an external cache, but the cache driver is not included in this PR.  I will create another PR for the external cache driver in drivers/cache/, and the missing reference ‘cache_data_range’ will also be added.  Before this, could you exclude the platform for the cache test?  Thank you very much.

--
Dylan

寄件者: eobalski ***@***.***>
日期: 星期四, 2022年3月31日 下午6:59
收件者: zephyrproject-rtos/zephyr ***@***.***>
副本: Dylan Hung ***@***.***>, Mention ***@***.***>
主旨: Re: [zephyrproject-rtos/zephyr] tests: Fix alignment (PR #44401)

@aspeeddylan<https://github.com/aspeeddylan> I saw you added
default CACHE_TYPE as HAS_EXTERNAL_CACHE<https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/aspeed/ast10x0/Kconfig.defconfig.series#L18:L20>. In this PR the test that uses cache management is run and it fails due to undefined reference to cache_data_range.

[...]/ast1030_evb/tests/subsys/ipc/icmsg_buf/subsys.ipc.icmsg_buf/../../../../../../../include/cache.h:91: undefined reference to `cache_data_range'

Could you please verify if HAS_EXTERNAL_CACHE should be selected for this particular soc?
It doesn't work with HAS_ARCH_CACHE either.
Could you please have a look? Should i exclude this platform from this test?

—
Reply to this email directly, view it on GitHub<https://github.com/zephyrproject-rtos/zephyr/pull/44401#issuecomment-1084421956>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALT4IUBMKHKAXGMUUQNXATLVCWAQHANCNFSM5SE35R3A>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

LixinGuoX commented:
Verified on intel_adsp_cavs18.

Commit:
tests: ipc: Update exclude platforms

remove qemu_xtensa from exclude platforms as fix for this platform
is provided.

add ast1030_evb to exclude platforms as this soc does not yet have
in tree driver for external cache controller.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

emob-nordic commented:
@LixinGuoX @nashif  could you have a look, accept and merge?

mbolivar-nordic commented on tests/lib/icmsg/src/main.c:
```diff
@@ -11,7 +11,7 @@
  * -1 because internal rd/wr_idx is reserved to mean the buffer is empty.
  */
-static uint8_t memory_area[212];
+static uint8_t memory_area[212] __aligned(4);
```
The commit log is just describing the changes without providing any justification for them. Why?

mbolivar-nordic replied:
Thanks for the updated commit log. I'm going to drop my NAK.

emob-nordic commented:
@mbolivar-nordic i updated the commit message, is that okay now?
